File: python/Trionic_Array_I_3637.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        if nums[0] >= nums[1]:
            return False
        
        Up, Down, Up2 = 0, 1, 2
        state = Up
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                return False
            if state == Up:
                if nums[i] < nums[i - 1]:
                    state = Down
            elif state == Down:
                if nums[i] > nums[i - 1]:
                    state = Up2
            else:
                if nums[i] < nums[i - 1]:
                    return False
        
        return state == Up2

File: cpp/Trionic_Array_I_3637.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        if (nums[0] >= nums[1]) return false; 
        enum State {Up, Down, Up2};
        State state = Up;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) return false;
            if (state == Up) {
                if (nums[i] < nums[i - 1]) state = Down;
            }
            else if (state == Down) {
                if (nums[i] > nums[i - 1]) state = Up2;
            }
            else {
                if (nums[i] < nums[i - 1]) return false;
            }
        }
        return (state == Up2);
    }
};
